libr/core/cmd.c

```diff
@@ -5953,6 +5983,356 @@ DEFINE_HANDLE_TS_FCN(iter_hit_command) {
 	return res;
 }
 
+DEFINE_HANDLE_TS_FCN(foreach_addrsize_command) {
+	RCore *core = state->core;
+	TSNode command = ts_node_named_child (node, 0);
+	uint32_t i = 1;
+	bool ret = true;
+	TSNode seek_addr_node = ts_node_named_child (node, i);
+	TSNode blk_sz_node = ts_node_named_child (node, i + 1);
+	while (!ts_node_is_null (seek_addr_node)) {
+		char *seek_addr_str = ts_node_handle_arg (state, node, seek_addr_node, i);
+		char *blk_sz_str = ts_node_handle_arg (state, node, blk_sz_node, i + 1);
+		ut64 seek_addr_val = r_num_get (NULL, seek_addr_str);
+		ut64 blk_sz_val = r_num_get (NULL, blk_sz_str);
+		free (seek_addr_str);
+		free (blk_sz_str);
+		r_core_seek (core, seek_addr_val, 1);
+		r_core_block_size (core, blk_sz_val);
+		ret &= handle_ts_command (state, command);
+		i += 2;
+		seek_addr_node = ts_node_named_child (node, i);
+		blk_sz_node = ts_node_named_child (node, i + 1);
+	}
+	return ret;
+}
+
+DEFINE_HANDLE_TS_FCN(foreach_bb_command) {
+	RCore *core = state->core;
+	TSNode command = ts_node_named_child (node, 0);
+	RAnalFunction *fcn = r_anal_get_fcn_in (core->anal, core->offset, 0);
+	ut64 offorig = core->offset;
+	ut64 obs = core->blocksize;
+	if (!fcn) {
+		return false;
+	}
+
+	RListIter *iter;
+	RAnalBlock *bb;
+	bool ret = true;
+	r_list_foreach (fcn->bbs, iter, bb) {
+		r_core_seek (core, bb->addr, 1);
```

Comment:
use true instead of 1, the arg expects bool

Reply:
good point... however I would feel a bit inconsistent in doing this here, as there are like ~250 uses of this function and probably ~200 use 0/1. Can I do a separate PR where I change ALL of them at once?

Reply:
+1 for the separate PR.